docs/parser.md

```diff
@@ -0,0 +1,247 @@
+
+# What is the parser?
+
+The parser component provided by this package is designed to make it easier to understand an existing JSON Schema, and
+to make it easier to provide clear feedback to end users.
+
+[JSON Schema](http://json-schema.org/) is a standard for describing JSON data formats using JSON notation.  Simple
+schemas are more or less human-readable descriptions of what structure is expected and what data is allowed at each
+level.
+
+You can reuse content within or between schemas using the `$ref` notation, which allows you to refer to content found
+elsewhere.  Schemas that make use of the `$ref` notation are not generally human readable, especially when there is a
+chain of inherited definitions that must each be retrieved and understood before you reach the level of "simple" rules.
+
+This component parses a JSON Schema and dereferences all of its `$ref` links (including chains of references).  The
+final result is a JSON Schema that enforces the same structure as the original, but which is composed only of "simple"
+rules.
+
+# What is it used for?
+
+The parser can be used to present a simplified (dereferenced schema) and to interrogate that schema for details about
+a given field.  Our first use case for both of these things is adding partial support for custom error messages as
+outlined in [draft v5 of the JSON Schema standard](https://github.com/json-schema/json-schema/wiki/Custom-error-messages-(v5-proposal)).
+
+# The v5 proposal
+
+The draft v5 standard adds an `errors` keyword, which is a map of relative JSON pointers and replacement message
+definitions.  It can be used either the document level, or when defining an individual field.  Here is an example of
+using `errors` at the document level, taken from the v5 draft proposal:
+
+```
+{
+  "properties": {
+    "age": { "minimum": 13 },
+    "gender": { "enum": ["male", "female"] }
+  },
+  "errors": {
+    "#/properties/age/minimum": "Should be at least ${schema} years, ${data} years is too young.",
```

Comment:
It's part of the JSON pointer standard, I added a short block explaining the two escaping conventions and linking to the standard.